samples/algorithms/SimpleVQE.qs

```diff
@@ -0,0 +1,200 @@
+/// To keep this sample simple hamiltonian terms are randomly selected.
+///
+/// # Reference
+/// Ground-state energy estimation of the water molecule on a trapped ion quantum
+/// computer by Yunseong Nam et al., 2019. https://arxiv.org/abs/1902.10171
+
+import Std.Arrays.IsEmpty;
+import Std.Arrays.IndexRange;
+import Std.Convert.IntAsDouble;
+import Std.Diagnostics.Fact;
+import Std.Math.AbsD;
+import Std.Math.PI;
+
+/// # Summary
+/// Find the approximation to the minimum eigenvalue of a Hamiltonian by applying VQE
+operation Main() : Double {
+
+    // Find the approximation to the minimum eigenvalue of a Hamiltonian
+    // by varying ansatz parameters to minimize its expectation value.
+    SimpleDescent(
+        // Use 1000 shots when estimating hamiltonian terms
+        FindHamiltonianExpectationValue(_, 100),
+        // Start from these angles for ansatz state preparation
+        [1.0, 1.0],
+        // Use initial step pi/8 to find minimum
+        PI() / 8.0,
+        // Stop optimization if step is 0
+        0.0,
+        // Stop optimization after 100 attempts to improve
+        100
+    )
+}
+
+/// # Summary
+/// Find expectation value of a Hamiltonian given parameters for the
+/// ansatz state and number of shots to evaluate each term.
+operation FindHamiltonianExpectationValue(thetas : Double[], shots : Int) : Double {
+    let terms = [
+        ([PauliZ, PauliI, PauliI, PauliI], 0.16),
+        ([PauliI, PauliI, PauliZ, PauliI], 0.25),
+        ([PauliZ, PauliZ, PauliI, PauliI], 0.17),
+        ([PauliI, PauliI, PauliZ, PauliZ], 0.45),
+        ([PauliX, PauliX, PauliX, PauliX], 0.2),
+        ([PauliY, PauliY, PauliY, PauliY], 0.1),
+        ([PauliY, PauliX, PauliX, PauliY], 0.02),
+        ([PauliX, PauliY, PauliY, PauliX], 0.22),
+    ];
+    mutable value = 0.0;
+    for (basis, coefficient) in terms {
+        value += coefficient * FindTermExpectationValue(thetas, basis, shots);
+    }
+    value
+}
+
+/// # Summary
+/// Find expectation value of a Hamiltonian term given parameters for the
+/// ansatz state, measurement basis and number of shots to evaluate each term.
+operation FindTermExpectationValue(
+    thetas : Double[],
+    pauliBasis : Pauli[],
+    shots : Int
+) : Double {
+
+    mutable zeroCount = 0;
+    for _ in 1..shots {
+        use qs = Qubit[4];
+        PrepareAnsatzState(qs, thetas);
+        if Measure(pauliBasis, qs) == Zero {
+            zeroCount += 1;
+        }
+        ResetAll(qs);
+    }
+    IntAsDouble(zeroCount) / IntAsDouble(shots)
+}
+
+/// # Summary
+/// Prepare the ansatz state for given parameters on a qubit register
+operation PrepareAnsatzState(qs : Qubit[], thetas : Double[]) : Unit {
+    BosonicExitationTerm(thetas[0], qs[0], qs[2]);
+    CNOT(qs[0], qs[1]);
+    NonBosonicExitataionTerm(thetas[1], qs[0], qs[1], qs[2], qs[3]);
+}
+
+/// # Summary
+/// Bosonic exitation circuit
+operation BosonicExitationTerm(
```

Comment:
It would be great to describe how this circuit was found or provide a reference that explains more.

Reply:
I updated the comment to mention referenced paper, which is referenced in the beginning of the sample.